src/uct/ib/base/ib_device.c

```diff
@@ -629,49 +634,54 @@ int uct_ib_device_test_roce_gid_index(uct_ib_device_t *dev, uint8_t port_num,
     return 1;
 }
 
-static ucs_status_t uct_ib_device_set_roce_gid_index(uct_ib_device_t *dev,
-                                                     uint8_t port_num,
-                                                     uint8_t *gid_index)
+ucs_status_t uct_ib_device_set_roce_info(uct_ib_device_t *dev, uint8_t port_num,
```

Comment:
rename to uct_ib_device_select_gid(dev, port_num, &gid_info) because this function performs the logic of selecting best gid according to roce version and IP version